### ISSUE TITLE
Trailing argument doesn't seem to work

### ISSUE THREAD

**Issue by Hanif-S** [casact/chainladder-python#456]
Hi,

I have settled claims data with date of loss starting from 1/1//2018 to 8/31/2023. I want to generate a triangle with partial origin exposure on the most recent accident year so the triangle rows would be AY2018, AY2019, AY2020, AY2021, AY2022, AY-YTD2023. I figured that by default, the trailing argument is True but when specify it to False in 2 different ways (as shown below) it still doesn't work.

**Method 1:**
T_SET = cl.Triangle(
            Actuarial_Data_SET,
            origin='DOL',
            development='Approved/Settled Date',
            columns=['Amt_Gross'],
            index='COB',
            cumulative=False,
            trailing=False
                ).grain('AYDY')

**Method 2:**
T_SET = cl.Triangle(
            Actuarial_Data_SET,
            origin='DOL',
            development='Approved/Settled Date',
            columns=['Amt_Gross'],
            index='COB',
            cumulative=False
                ).grain('AYDY', trailing=False)

Im not sure if this is relevant, but it looks the trailing condition doesnt function here
![image](https://github.com/casact/chainladder-python/assets/144644034/6e1e3b4a-dcac-429e-8892-7f7ca05ed729)

Thanks in advance!

**Comment by jbogaardt** [comment]
One or the other (ideally both) approaches should work, so investigating the issue.

**Comment by jbogaardt** [comment]
Reproducible example:

```python
import chainladder as cl
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/casact/chainladder-python/pl_tri/chainladder/utils/data/prism.csv')

# Trailing in constructor doesn't work
assert (cl.Triangle(
    df[pd.to_datetime(df['PaymentDate'])<'2017-08-01'],
    origin='AccidentDate', development='PaymentDate', 
    columns=['Paid'], cumulative=False, trailing=True
).grain('OYDM') != 
cl.Triangle(
    df[pd.to_datetime(df['PaymentDate'])<'2017-08-01'],
    origin='AccidentDate', development='PaymentDate', 
    columns=['Paid'], cumulative=False, trailing=False
).grain('OYDM'))

# trailing in grain doesn't work
assert (cl.Triangle(
    df[pd.to_datetime(df['PaymentDate'])<'2017-08-01'],
    origin='AccidentDate', development='PaymentDate', 
    columns=['Paid'], cumulative=False, 
).grain('OYDM', trailing=True) != 
cl.Triangle(
    df[pd.to_datetime(df['PaymentDate'])<'2017-08-01'],
    origin='AccidentDate', development='PaymentDate', 
    columns=['Paid'], cumulative=False, 
).grain('OYDM', trailing=False))
```

**Comment by jbogaardt** [comment]
@Hanif-S, indeed you are correct. I've modified that line to work again. I still think I need to test variations of the `trailing` argument in the constructor as well, but the I believe the `grain` function should be working appropriately now.

**Comment by jbogaardt** [comment]
Released in v0.8.18
